### PR TITLE
Fix hdfs reading from files with spaces

### LINKED ISSUE #13606: Пустой результат для select, если каталог в HDFS содержит пробел в имени
**Describe the bug**
Если в имени одного из каталогов в URI содержится пробел, то CH не выдаёт никаких ошибок при создании таблицы и выборке из неё, но на запрос возвращает пустой набор данных.

В каталоге hdfs://192.168.1.100:9000/folder1/folder2/DATE=2020-08-01/ находятся каталоги вида:

- PROCESSING_DT=2020-08-02 02%3A41%3A25.176
- PROCESSING_DT=2020-08-03 03%3A21%3A38.336

Внутри каталогов PROCESSING_DT=* находятся файлы Parquet. При создании таблицы всё проходит без ошибок. При запуске запроса select тоже никаких ошибок, но данных нет. Если, например, заменить пробел в имени каталога на символ подчёркивания ("PROCESSING_DT=2020-08-02_02%3A41%3A25.176"), то всё начинает работать как ожидается и запрос select возвращает данные.

**How to reproduce**
* ClickHouse server version
20.5.2.7

* `CREATE TABLE` statements for all tables involved
drop table if exists hdfs.test_table;
create table hdfs.test_table
(
    REC_ID UInt64,
    REC_NAME String,
    VALUE Nullable(String)
)
engine = HDFS('hdfs://192.168.1.100:9000/folder1/folder2/DATE=2020-08-01/*/*/', 'Parquet');

* Sample data for all these tables
Любые данные

* Queries to run that lead to unexpected result
select * from hdfs.test_table limit 5;

**Expected behavior**
Получение данных из файлов Parquet

### PR DESCRIPTION
**Pull request by kssenii**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Closes https://github.com/ClickHouse/ClickHouse/issues/13606.

Checked locally with the following files:
> bash-4.1# hdfs dfs -put kssenii.txt '/test/test test test 0.txt'
bash-4.1# hdfs dfs -put kssenii.txt '/test/test test test 1.txt'
bash-4.1# hdfs dfs -put kssenii.txt '/test/test test test 2.txt'
bash-4.1# hadoop fs -ls /test
Found 3 items
-rw-r--r-- 1 root supergroup 292 2021-04-17 07:31 /test/test test test 0.txt
-rw-r--r-- 1 root supergroup 292 2021-04-17 07:31 /test/test test test 1.txt
-rw-r--r-- 1 root supergroup 292 2021-04-17 07:31 /test/test test test 2.txt

And this query now works:
>  CREATE TABLE test ( id UInt32 ) ENGINE = HDFS('hdfs://192.168.112.2:9000/test/test*', 'TabSeparated');
select * from test;

Fix: It is enough to pass to hdfsBuilder uri without file path. File path is handled separately in hdfsOpenFile method. It should not be passed to hdfsBuilder because at the moment of builder creation it is not url-encoded.

### CODE REVIEW

File: src/Storages/HDFS/ReadBufferFromHDFS.cpp [hunk 1]
```diff
@@ -66,8 +70,12 @@ ReadBufferFromHDFS::ReadBufferFromHDFS(const std::string & hdfs_name_,
     const Poco::Util::AbstractConfiguration & config_,
     size_t buf_size_)
     : BufferWithOwnMemory<ReadBuffer>(buf_size_)
-    , impl(std::make_unique<ReadBufferFromHDFSImpl>(hdfs_name_, config_))
 {
+    const size_t begin_of_path = hdfs_name_.find('/', hdfs_name_.find("//") + 2);
```
Comment (alexey-milovidov) [inline src/Storages/HDFS/ReadBufferFromHDFS.cpp:74]: Need to add more checks.
Check that '//' has been found.
Check that the next '/' has been found.

Or use Poco::URI for this task?
The file name with spaces should be URL-encoded by the user.

**Review by alexey-milovidov — COMMENTED** [review]
TODO: add integration test.

File: src/Storages/HDFS/ReadBufferFromHDFS.cpp (hunk 1 above)
Comment (kssenii) [reply src/Storages/HDFS/ReadBufferFromHDFS.cpp:74]: > Check that the next '/' has been found.

There is already  a check for user-defined url [here](https://github.com/ClickHouse/ClickHouse/blob/master/src/Storages/HDFS/StorageHDFS.cpp#L51).

> Check that '//' has been found.

It is not necessarily present.

Comment (alexey-milovidov) [reply src/Storages/HDFS/ReadBufferFromHDFS.cpp:74]: If `//` is not necessarily present, what does `hdfs_name_.find("//") + 2` mean?

Comment (alexey-milovidov) [reply src/Storages/HDFS/ReadBufferFromHDFS.cpp:74]: > There is already a check for user-defined url here.

But it means you can use Poco::URI in this place also.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**